pages/dao/[symbol]/proposal/components/instructions/UXD/DepositInsuranceToMangoDepository.tsx

```diff
@@ -1,166 +1,74 @@
+import SelectOptionList from '../../SelectOptionList'
 
-const DepositInsuranceToMangoDepository = ({
+const UXDDepositInsuranceToMangoDepository = ({
   index,
-  governance,
+  governedAccount,
 }: {
   index: number
-  governance: ProgramAccount<Governance> | null
+  governedAccount: GovernedMultiTypeAccount | undefined
 }) => {
-  const connection = useWalletStore((s) => s.connection)
-  const wallet = useWalletStore((s) => s.current)
-  const { realmInfo } = useRealm()
-  const { getGovernancesByAccountTypes } = useGovernanceAssets()
-  const governedProgramAccounts = getGovernancesByAccountTypes([
-    GovernanceAccountType.ProgramGovernanceV1,
-    GovernanceAccountType.ProgramGovernanceV2,
-  ]).map((x) => {
-    return {
-      governance: x,
-    }
-  })
-  const shouldBeGoverned = index !== 0 && governance
-  const programId: PublicKey | undefined = realmInfo?.programId
-  const [form, setForm] = useState<DepositInsuranceToMangoDepositoryForm>({
-    governedAccount: undefined,
-    programId: programId?.toString(),
-    collateralName: '',
-    insuranceName: '',
-    insuranceDepositedAmount: 0,
-  })
-  const [formErrors, setFormErrors] = useState({})
-  const { handleSetInstructions } = useContext(NewProposalContext)
-  const handleSetForm = ({ propertyName, value }) => {
-    setFormErrors({})
-    setForm({ ...form, [propertyName]: value })
-  }
-  const validateInstruction = async (): Promise<boolean> => {
-    const { isValid, validationErrors } = await isFormValid(schema, form)
-    setFormErrors(validationErrors)
-    return isValid
-  }
-  async function getInstruction(): Promise<UiInstruction> {
-    const isValid = await validateInstruction()
-    let serializedInstruction = ''
-    if (
-      isValid &&
-      programId &&
-      form.governedAccount?.governance?.account &&
-      wallet?.publicKey
-    ) {
-      const createIx = await createDepositInsuranceToMangoDepositoryInstruction(
+  const {
+    connection,
+    form,
+    formErrors,
+    handleSetForm,
+  } = useInstructionFormBuilder<DepositInsuranceToMangoDepositoryForm>({
+    index,
+    initialFormValues: {
+      governedAccount,
+      insuranceDepositedAmount: 0,
+    },
+    schema: yup.object().shape({
+      collateralName: yup
+        .string()
+        .required('Collateral Name address is required'),
+      insuranceName: yup
+        .string()
+        .required('Insurance Name address is required'),
+      insuranceDepositedAmount: yup
+        .number()
+        .moreThan(0, 'Insurance Deposited amount should be more than 0')
+        .required('Insurance Deposited amount is required'),
+      governedAccount: yup
+        .object()
+        .nullable()
+        .required('Program governed account is required'),
+    }),
+    buildInstruction: async function () {
+      if (!governedAccount?.governance?.account) {
+        throw new Error('Governance must be a Program Account Governance')
+      }
+      if (
+        !form.collateralName ||
```

Comment:
Either we use the `if`, either we use `!` ^^.

I would use `!` as you did for `form.governedAccount!.governance!.pubkey` because we know the values cannot be _undefined_.